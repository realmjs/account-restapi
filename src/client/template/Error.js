"use strict"

import React, { Component } from 'react'

export default class Error extends Component {
  constructor(props) {
    super(props)
  }
  componentDidMount() {
    const error = this.props.data.error
    this.props.done({ status: error.code })
  }
  render() {
    const error = this.props.data.error
    return (
      <div className="w3-container" style={{ padding: "10px 12px", maxWidth: "460px" }}>
        <header>
          <span onClick={this.props.close} className="w3-button w3-right w3-red"> &times; </span>
        </header>
        <div>
          <h4 className="w3-text-red"> {error.code} </h4>
          <p> {error.detail} </p>
        </div>
      </div>
    )
  }
}
