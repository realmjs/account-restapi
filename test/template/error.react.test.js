"use strict"

import React from 'react';
import renderer from 'react-test-renderer';
import {screen, fireEvent, render} from '@testing-library/react';

import Error from '../../src/client/Template/Error';

test('Should match snapshot and call done when mounted', () => {
  const done = jest.fn(msg => msg);
  const error = {
    code: 400, detail: 'Bad Request',
  }
  const component = renderer.create(
    <Error data = {{ error }} done = {done} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(done).toHaveBeenCalled();
  expect(done.mock.results[0].value).toEqual({status: 400});
});


test('Should call close when click to <span>x</span>', () => {
  const done = jest.fn(msg => msg);
  const close = jest.fn();
  const error = {
    code: 400, detail: 'Bad Request',
  }
  render(
    <Error data = {{ error }} done = {done} close = {close} />
  );
  fireEvent.click(screen.getByText('\u00D7'));  // \u00D7 is unicode of 'Ã—'
  expect(close).toHaveBeenCalled();
});
