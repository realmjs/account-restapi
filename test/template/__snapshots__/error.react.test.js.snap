// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot and call done when mounted 1`] = `
<div
  className="w3-container"
  style={
    Object {
      "maxWidth": "460px",
      "padding": "10px 12px",
    }
  }
>
  <header>
    <span
      className="w3-button w3-right w3-red"
    >
       Ã— 
    </span>
  </header>
  <div>
    <h4
      className="w3-text-red"
    >
       
      400
       
    </h4>
    <p>
       
      Bad Request
       
    </p>
  </div>
</div>
`;
